.github/workflows/create-azure-self-hosted-runners.yml

```diff
@@ -0,0 +1,119 @@
+name: create-azure-self-hosted-runners
+
+on:
+  workflow_dispatch:
+    inputs:
+      amount_of_runners:
+        description: 'Amount of runners to set up'
+        required: true
+      runner_scope:
+        type: choice
+        required: true
+        description: Scope of the runner. On personal accounts, only "repo-level" works
+        options:
+        - org-level
+        - repo-level
+      runner_org:
+        type: string
+        required: true
+        description: Organization or personal account to deploy the runner to
+      runner_repo:
+        type: string
+        required: false
+        description: Repo to deploy the runner to. Only needed if runner_scope is set to "repo-level"
+
+env:
+  ACTIONS_RUNNER_ORG: "${{ github.event.inputs.runner_org }}"
+  ACTIONS_RUNNER_REPO: "${{ github.event.inputs.runner_repo }}"
+  # This has to be a public URL that the VM can access after creation
+  POST_DEPLOYMENT_SCRIPT_URL: https://raw.githubusercontent.com/git-for-windows/git-for-windows-automation/main/azure-self-hosted-runners/post-deployment-script.ps1
+
+# The following secrets are required for this workflow to run:
+# GH_API_PAT - PAT to create a runner registration token using GitHub's API
+# AZURE_CREDENTIALS - Credentials for the Azure CLI. It's recommended to set up a resource
+#                     group specifically for self-hosted Actions Runners.
+#   az ad sp create-for-rbac --name "{YOUR_DESCRIPTIVE_NAME_HERE}" --role contributor \
+#     --scopes /subscriptions/{SUBSCRIPTION_ID_HERE}/resourceGroups/{RESOURCE_GROUP_HERE} \
+#     --sdk-auth
+# AZURE_RESOURCE_GROUP - Resource group to create the runner(s) in
```

Comment:
I assumed that you want to create these VMs in their own, dedicated resource group. This pipeline expects the resource group to be created in Azure prior to running the pipeline.

Reply:
That makes sense. Then we can delete the resource group which should delete everything contained therein.